libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/sfhybrid/SalesforceDroidGapActivity.java

```diff
@@ -131,31 +131,17 @@ protected ClientManager buildClientManager() {
 	}
 
 	@Override
-	public void init()
-	{
+	public void init() {
     	Log.i("SalesforceDroidGapActivity.init", "init called");
     	super.init();
-    	final String uaStr = SalesforceSDKManager.getInstance().getUserAgent();
-    	if (null != this.appView) {
-    		WebSettings webSettings = this.appView.getSettings();
-    		String origUserAgent = webSettings.getUserAgentString();
-    		final String extendedUserAgentString = uaStr + " Hybrid " + (origUserAgent == null ? "" : origUserAgent);
-    		webSettings.setUserAgentString(extendedUserAgentString);
-
-    		// Configure HTML5 cache support.
-    		webSettings.setDomStorageEnabled(true);
-    		String cachePath = getApplicationContext().getCacheDir().getAbsolutePath();
-    		webSettings.setAppCachePath(cachePath);
-    		webSettings.setAppCacheEnabled(true);
-    		webSettings.setAllowFileAccess(true);
-    		webSettings.setCacheMode(WebSettings.LOAD_DEFAULT);
-    		EventsObservable.get().notifyEvent(EventType.GapWebViewCreateComplete, appView);
-      	}
+    	EventsObservable.get().notifyEvent(EventType.GapWebViewCreateComplete, appView);
 	}
 
 	@Override
-    protected CordovaWebViewClient makeWebViewClient(CordovaWebView webView) {
-        return new SalesforceIceCreamWebViewClient(this, webView);
+    protected CordovaWebViewEngine makeWebViewEngine() {
```

Comment:
This is where most of the interfaces had to be changed and where the meat of the upgrade is.